ARGO/src/AiSystem.cpp

```diff
@@ -58,3 +65,46 @@ void AiSystem::simpleMoveAi(TransformComponent* t_posComp)
 		t_posComp->m_movingUp = false;
 	}
 }
+
+void AiSystem::meleeAI(TransformComponent* t_posComp, AiComponent* t_aiComponent)
+{
+	//The Only Possible States Available to Melee Enemies. Use to limit Behaviours
+	switch (t_aiComponent->getStates()) {
+	case eSleeping:
+		sleep(t_posComp, t_aiComponent);
+		break;
+	case eWander:
+		wander(t_posComp,t_aiComponent);
+	}
+}
+
+void AiSystem::rangedAI(TransformComponent* t_posComp, AiComponent* t_aiComponent)
+{
+	//The Only Possible States Available to Ranged Enemies. Use to limit Behaviours
+	switch (t_aiComponent->getStates()) {
+	case eSleeping:
+		sleep(t_posComp, t_aiComponent);
+		break;
+	case eWander:
+		wander(t_posComp, t_aiComponent);
+	}
+}
+
+void AiSystem::wander(TransformComponent* t_posComp, AiComponent* t_aiComponent)
+{
+	float tempAdjuster = (((rand() % 20 + 1)));//Gives a number between 1 and 20
+	tempAdjuster -= 10; //-10 to give a range of -9 and 10,
```

Comment:
why not just get random number between -9 and 10 instead of 1-20.
Also this can be done on the line above

Reply:
same on line below

Reply:
rand doesn't give negative values.